include/plat/spike/plat/machine/hardware.h

```diff
@@ -31,10 +31,15 @@
 #define PADDR_BASE 0x0lu
 #endif
 
+#ifndef CONFIG_BUILD_ROCKET_CHIP
```

Comment:
Having this line (i.e. the _#ifndef_) priorities _CONFIG_BUILD_ROCKET_CHIP_ over any other, future, platforms. I'd suggest you _#ifdef_ for each specific platform. i.e.:

```
#ifdef CONFIG_BUILD_ROCKET_CHIP

#elif CONFIG_PLAT_SPIKE

#elif CONFIG_PLAT_XXX

#else 
#error "Not able to determine a platform
#endif
```

Reply:
Thats a good point about the #ifndef wrapper. The way we structured the ROCKET_CHIP option was to have it be an addition to the spike platform instead of its own unique platform, since so much of the code is identical. 

So in the gen_config.h file, CONFIG_PLAT_SPIKE is 1 as well as the CONFIG_BUILD_ROCKET_CHIP.

The way you have it could work, since the CONFIG_PLAT_SPIKE line wouldn't get processed, but this file would only get compiled if PLAT_SPIKE was selected.

I'll update it to use #ifdefs instead of #ifndefs